src/components/Pressable/Pressable.tsx

```diff
@@ -183,42 +243,90 @@ export default function Pressable(props: PressableProps) {
         })
         .onTouchesUp((event) => {
           if (handlingOnTouchesDown.current) {
-            onEndHandlingTouchesDown.current = () => pressOutHandler(event);
+            onEndHandlingTouchesDown.current = () =>
+              pressOutHandler(gestureTouchToPressableEvent(event));
             return;
           }
-
-          pressOutHandler(event);
+          // On iOS, when pressed out before Native gesture pressed in,
+          // without setting this variable Native gesture would think it's the first one to be launched
```

Comment:
Is there a way to somehow rephrase this sentence? It is not the easiest one to read 😅

Reply:
Oh yeah I see the issue 😄 
Do you think this would work?
```suggestion
          // On iOS, short taps will make `LongPress` gesture call `onTouchesUp` before `Native` gesture calls  `onStart` 
          // This variable makes sure that `onStart` is not detected as the first gesture since the `Pressable` was pressed.
```

Reply:
Yeah, that's better. I'd change _**"makes sure"**_ to _**"ensures"**_ though